packages/catalog-process/src/services/catalogService.ts

```diff
@@ -971,209 +886,220 @@ export function catalogServiceBuilder(
-    throw notValidDescriptor(descriptor.id, descriptor.state.toString());
-  }
+      return updatedEService;
+    },
 
-  if (descriptor.interface === undefined) {
-    throw eServiceDescriptorWithoutInterface(descriptor.id);
-  }
+    async publishDescriptor(
+      eserviceId: EServiceId,
+      descriptorId: DescriptorId,
+      authData: AuthData
+    ): Promise<void> {
+      logger.info(
+        `Publishing Descriptor ${descriptorId} for EService ${eserviceId}`
+      );
 
-  const currentActiveDescriptor = eservice.data.descriptors.find(
-    (d: Descriptor) => d.state === descriptorState.published
-  );
+      const eservice = await retrieveEService(eserviceId, readModelService);
+      assertRequesterAllowed(eservice.data.producerId, authData.organizationId);
 
-  const updatedDescriptor = updateDescriptorState(
-    descriptor,
-    descriptorState.published
-  );
+      const descriptor = retrieveDescriptor(descriptorId, eservice);
+      if (descriptor.state !== descriptorState.draft) {
+        throw notValidDescriptor(descriptor.id, descriptor.state.toString());
+      }
 
-  const newEservice = updateDescriptor(eservice.data, updatedDescriptor);
+      if (descriptor.interface === undefined) {
+        throw eServiceDescriptorWithoutInterface(descriptor.id);
+      }
 
-  if (currentActiveDescriptor !== undefined) {
-    const newEserviceWithDeprecation = updateDescriptor(
-      eservice.data,
-      deprecateDescriptor(eserviceId, currentActiveDescriptor)
-    );
+      const currentActiveDescriptor = eservice.data.descriptors.find(
+        (d: Descriptor) => d.state === descriptorState.published
+      );
 
-    return toCreateEventEServiceDescriptorPublished(
-      eserviceId,
-      eservice.metadata.version + 1,
-      descriptorId,
-      newEserviceWithDeprecation
-    );
-  } else {
-    return toCreateEventEServiceDescriptorPublished(
-      eserviceId,
-      eservice.metadata.version,
-      descriptorId,
-      newEservice
-    );
-  }
-}
+      const updatedDescriptor = updateDescriptorState(
+        descriptor,
+        descriptorState.published
+      );
 
-export function suspendDescriptorLogic({
-  eserviceId,
-  descriptorId,
-  authData,
-  eservice,
-}: {
-  eserviceId: EServiceId;
-  descriptorId: DescriptorId;
-  authData: AuthData;
-  eservice: WithMetadata<EService> | undefined;
-}): CreateEvent<EServiceEvent> {
-  assertEServiceExist(eserviceId, eservice);
-  assertRequesterAllowed(eservice.data.producerId, authData.organizationId);
+      const newEservice = updateDescriptor(eservice.data, updatedDescriptor);
 
-  const descriptor = retrieveDescriptor(descriptorId, eservice);
-  if (
-    descriptor.state !== descriptorState.deprecated &&
-    descriptor.state !== descriptorState.published
-  ) {
-    throw notValidDescriptor(descriptorId, descriptor.state.toString());
-  }
+      // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
We have this eslint rule disabled in various places maybe we could decide to disable it in the entire project